tutorials/llm/llama-3/nemo2-sft-peft/nemo2-sft.ipynb

```diff
@@ -0,0 +1,632 @@
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def adam_with_cosine_annealing() -> run.Config[nl.OptimizerModule]:\n",
+    "    opt_cfg = run.Config(\n",
+    "        OptimizerConfig,\n",
+    "        optimizer=\"adam\",\n",
+    "        lr=5e-6,\n",
+    "        adam_beta2=0.98,\n",
+    "        use_distributed_optimizer=True,\n",
+    "        clip_grad=1.0,\n",
+    "        bf16=True,\n",
+    "    )\n",
+    "    return run.Config(\n",
+    "        nl.MegatronOptimizerModule,\n",
+    "        config=opt_cfg\n",
+    "    )\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Base Model\n",
+    "We will perform SFT on top of Llama3-8b so we create a `LlamaModel` to pass to finetune API."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def llama3_8b() -> run.Config[pl.LightningModule]:\n",
+    "    return run.Config(llm.LlamaModel, config=run.Config(llm.Llama3Config8B))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### AutoResume\n",
+    "In NeMo 2.0 we can directly pass in Llama3-8b's Hugging Face ID to start SFT without manually converting it into NeMo checkpoint format like in NeMo 1.0."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def resume() -> run.Config[nl.AutoResume]:\n",
+    "    return run.Config(\n",
+    "        nl.AutoResume,\n",
+    "        restore_config=run.Config(nl.RestoreConfig,\n",
+    "            path=\"nemo://meta-llama/Meta-Llama-3-8B\"\n",
+    "        ),\n",
+    "        resume_if_exists=True,\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "### NeMo 2.0 finetun API\n",
+    "Using all the components we created above, we can call NeMo 2.0 finetun API:\n",
+    "```\n",
+    "llm.finetune(\n",
+    "    model=llama3_8b(),\n",
+    "    data=dolly(),\n",
+    "    trainer=trainer(),\n",
+    "    log=logger(),\n",
+    "    optim=adam_with_cosine_annealing(),\n",
+    "    resume=resume(),\n",
+    ")\n",
+    "```\n",
+    "Below is a python script that you can save as a file e.g. `nemo2-sft.py`, and run SFT training, using all components we created above and NeMo 2.0 finetune API. The script cannot be directly executed in interactive environment like a notebook. We can execute by `torchrun --nproc_per_node=<NUM_GPU> nemo2-sft.py` when multiple GPU is used."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def configure_finetuning_recipe():\n",
+    "    return run.Partial(\n",
```

Comment:
How does the user know if the training run is succesful? WandB graphs? perplexity score on validation set?

Reply:
NeMo-Run has a "SUCCEED" image/banner shown in the output cell when the run ends.